### ISSUE TITLE
Noticing lower performance with LeftWM compared to other window managers.

### ISSUE THREAD

**Issue by Joe23232** [leftwm/leftwm#108]
So I am running ArchLabs (based on Arch Linux) on virtualbox, I have allocated 2 Cores of CPU and 2048 MB of RAM.

While using a DE such as XFCE, any terminal programs lag when typing in text. But when I log into `awesome` WM and `spectrwm`, any terminal application that I use, inputing text is a lot smoother. but on `leftwm` terminal applications are a little bit laggy with text input.

**Comment by lex148** [comment]
Very interesting. This is a little unusual. LeftWM should be behaving quite performant.  

If you have a minute I would like to get a little more info about your setup. 
1) Did you compile LeftWM from source? if so did you use the --release flag?
2) Are you running a compositor such as compton or Picom? (LeftWM was really made to run in parallel with an external compositor)
3) Are you booting leftwm with `exec dbus-launch leftwm`? Without dbus you will get all kinds of lag and timeouts. 
4) Not every terminal emulator talks to Xorg in the same way. LeftWM could be doing something funky here. What terminal emulator are you using?

**Comment by Joe23232** [comment]
> Did you compile LeftWM from source? if so did you use the --release flag?

Yes I compiled from source I belive as I installed it from the AUR and no I did not use the `--release` flag.

> Are you running a compositor such as compton or Picom? (LeftWM was really made to run in parallel with an external compositor)

I ran `htop` and tried to search `compton` and `picom` but they don't seem to be running.

> Are you booting leftwm with exec dbus-launch leftwm? Without dbus you will get all kinds of lag and timeouts.

To be honest I have no idea, how would I check that, I am not too technical here :/

> Not every terminal emulator talks to Xorg in the same way. LeftWM could be doing something funky here. What terminal emulator are you using?

I am using `alacritty` but even `alacritty` was running fine on `spectrwm` and `awesome`.

**Comment by lex148** [comment]
If you installed from AUR most of these issues should be avoided. 
I would recommend installing a compositor such as [Picom](https://wiki.archlinux.org/index.php/Picom).
```
sudo pacman -S picom
```

All of the basic default themes are setup to use one.


It has been a while sense I have run LeftWM inside virtualbox. If this isn't the issue please let me know, I will give it a go when I get home.

**Comment by Joe23232** [comment]
I will try picom but wouldn't using a compositor just slow things down?
